google/cloud/spanner/internal/polling_loop_test.cc

```diff
@@ -0,0 +1,254 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/spanner/internal/polling_loop.h"
+#include "google/cloud/testing_util/assert_ok.h"
+#include <google/protobuf/struct.pb.h>
+#include <gmock/gmock.h>
+
+namespace google {
+namespace cloud {
+namespace spanner {
+inline namespace SPANNER_CLIENT_NS {
+namespace internal {
+namespace {
+
+using ::testing::HasSubstr;
+
+std::unique_ptr<PollingPolicy> TestPollingPolicy() {
+  using Policy = GenericPollingPolicy<LimitedErrorCountRetryPolicy,
+                                      ExponentialBackoffPolicy>;
+  return Policy(LimitedErrorCountRetryPolicy(5),
+                ExponentialBackoffPolicy(std::chrono::microseconds(1),
+                                         std::chrono::microseconds(5), 2.0))
+      .clone();
+}
+
+TEST(PollingLoopTest, ImmediateSuccess) {
+  google::protobuf::Value expected;
+  expected.set_string_value("42");
+
+  google::longrunning::Operation operation;
+  operation.set_name("test-operation");
+  operation.set_done(true);
+  operation.mutable_response()->PackFrom(expected);
+
+  StatusOr<google::protobuf::Value> actual =
+      PollingLoop<google::protobuf::Value>(
+          TestPollingPolicy(),
+          [](grpc::ClientContext&,
+             google::longrunning::GetOperationRequest const&) {
+            // The polling operation should not be called for requests
+            EXPECT_TRUE(false);
```

Comment:
You may be able to use the `FAIL()` googletest macro here to make this error case more obvious when visually scanning the file.

Reply:
`FAIL()` (and `GTEST_FAIL()`) have an embedded `return;` (note the semi-colon), so they do not work here.  Is there a `FAIL_WITH_RETURN()` or something?  I have never been able to find an alternative (ditto for `ASSERT_*()`).